acts: add v36.0.0, v36.1.0 and fixes

stephenswat opened the pull request:
This commit makes several changes to the Acts repository, namely:

1. It adds versions 36.0.0 and 36.1.0.
2. It adds the traccc plugin and related dependencies.
3. It updates the version requirements of some dependencies.
4. It adds the Geant4 module of GeoModel.
5. It updates the C++ standard requirement.
6. It adds a new variant determining the scalar type to use.

This commit supersedes #45851. Thanks @jmcarcell for the version updates; I have added you as co-author.

stephenswat committed:
acts: add v36.0.0, v36.1.0 and fixes

This commit makes several changes to the Acts repository, namely:

1. It adds versions 36.0.0 and 36.1.0.
2. It adds the traccc plugin and related dependencies.
3. It updates the version requirements of some dependencies.
4. It adds the Geant4 module of GeoModel.
5. It updates the C++ standard requirement.
6. It adds a new variant determining the scalar type to use.

This commit supercedes #45851. Thanks @jmcarcell for the version
updates; I have added you as co-author.

Co-authored-by: Juan Miguel Carceller <22276694+jmcarcell@users.noreply.github.com>

wdconinc commented on var/spack/repos/builtin/packages/acts/package.py:
```diff
@@ -186,7 +188,11 @@ class Acts(CMakePackage, CudaPackage):
+    )
     variant(
         "cxxstd",
         default="17",
```
Does this still work?? `_cxxstd_values` now doesn't have 17, but the default is set to 17. What does it pick as default in practice? I though this is one of those cases where we have to duplicate the variant directive with different when clauses to have the default change accordingly.

wdconinc commented on var/spack/repos/builtin/packages/acts/package.py:
```diff
@@ -219,6 +225,14 @@ class Acts(CMakePackage, CudaPackage):
         description="Log level above which examples should auto-crash",
     )
+    variant(
+        "scalar",
```
Does the new scalar variant not need to be set on cmake configuration?

stephenswat replied:
Yes, of course :man_facepalming:. Well spotted!

stephenswat commented:
I've updated this PR with the following new changes:

1. Split the variant declaration in two, one for C++17-standard versions and one for C++20-standard versions.
1. Added the scalar type to the CMake arguments.
1. Correctly passed on the scalar type to the detray dependency.
1. Removed the logic for checking if C++ standards as conditional, as they all are now.
1. Added a conflict with C++20 and gcc <= 9.
1. Added myself as maintainer.

Thanks @wdconinc

stephenswat committed:
Updates

wdconinc commented on var/spack/repos/builtin/packages/acts/package.py:
```diff
@@ -463,6 +484,8 @@ def plugin_cmake_variant(plugin_name, spack_variant):
             example_cmake_variant("TBB", "tbb", "USE"),
             cmake_variant(unit_tests_label, "unit_tests"),
+            plugin_cmake_variant("TRACCC", "traccc"),
+            self.define_from_variant("ACTS_CUSTOM_SCALARTYPE", "scalar"),
```
Minor. Move up so it is alphabetically sorted based on `scalar`.

stephenswat replied:
Done!

wdconinc reviewed (commented):
One minor stylistic request. Then ready for merging.

stephenswat committed:
alphabetic cmake args
